pkg/testcase/secretsencrypt.go

```diff
@@ -47,6 +47,8 @@ func secretsEncryptOps(action, product, cpIP string, nodes []shared.Node) {
 	if (action == "reencrypt") || (action == "rotate-keys") {
 		shared.LogLevel("DEBUG", "reencrypt op needs some time to complete - Sleep for 20 seconds before service restarts")
 		time.Sleep(20 * time.Second) // Wait for reencrypt action to complete before restarting services.
+	} else {
```

Comment:
instead adding more sleeps, can we refactor secrets encrypts to use k8s pkg?

Reply:
we have a lot of places here that we can remove all sleeps and add validations/waits  from there

Reply:
We need some minimum sleeps - since the actual prepare/rotate/reencrypt ops happen in the background - there is no quantifier/verifier for this - and the validation/waits of pods/nodes are not representative of this. This minimum wait is unavoidable. We have checks for pods/nodes as well.

Reply:
One way can be to check for status of `reencrypt` or `rotate_keys`.. once they have status of `finished` then we can proceed which can help with removing this hard coded sleeps

Reply:
no - we need sleeps before we restart the servers and in between restart of servers - during this time, the status wont say finished. only post restart of the servers will the status say finished. so these sleeps are unavoidable. we do have test for checking the status of reencrypt and rotate-keys to finished in the logEncryptionFileContents method.  and the secrets-encrypt status output.